frontend/src/Sidebar.tsx

```diff
@@ -0,0 +1,99 @@
+import React from "react";
+import './Sidebar.css';
+import { LoginManager } from "./LoginManager";
+
+export interface PageComponents {
+    components: PageComponent[]
+    loginManager: LoginManager
+}
+
+export interface PageComponent {
+    id: number
+    title: string
+    page: JSX.Element
+    icon: JSX.Element
+    selected?: boolean
+}
+
+/** Menu component which holds the different pages and makes them accesible. */
+export class Sidebar extends React.Component<PageComponents, PageComponents>{
+    constructor(props: PageComponents, state: PageComponents) {
+        super(props);
+        this.state = props;
+    }
+
+    /** Set this page active. Remove the old page and select the new one to be rendered. */
+    private setActivePage(pageId: number) {
+        var newComponents = [];
+        for (let c = 0; c < this.state.components.length; c++) {
+            let comp = this.state.components[c];
+            if (comp.selected && comp.id !== pageId) {
+                newComponents.push({ id: comp.id, title: comp.title, page: comp.page, icon: comp.icon })
+            }
+            else if (comp.id === pageId) {
+                newComponents.push({ id: comp.id, title: comp.title, page: comp.page, icon: comp.icon, selected: true })
+            } else {
+                newComponents.push(comp);
+            }
+        }
+
+        this.setState({ components: newComponents });
+        //TODO: set title and url in browser window
+    }
+
+    /** @return the header component with the titleo f the page. */
+    private getTitleComponent(elem?: PageComponent) {
+        if (elem) {
+
+            return (
+                <div>
+                    {elem?.icon}
+                    <div className="navbar-text">{elem?.title}</div>
+                </div>
+            );
+        } else {
+            return <div />;
+        }
+    }
+
+    public render() {
+        //Create the menu items
+        const components = this.state.components.map((elem) => {
+            const clazz = elem.selected ? 'navbar-menuItem navbar-menuItem-active' : 'navbar-menuItem';
+            return (
+                <a onClick={() => this.setActivePage(elem.id)}>
+                    <div className={clazz}>
+                        {elem.icon}
+                        <div className="navbar-text">{elem.title}</div>
+                    </div>
+                </a>
+            );
+        });
+
+        const selectedComponent = this.state.components.find((c) => { return c.selected });
```

Comment:
Might be simplified to
```suggestion
        const selectedComponent = this.state.components.find(c => c.selected);
```

Reply:
done